x/tbr/genesis.go

```diff
@@ -2,48 +2,46 @@ package tbr
 
 import (
 	"errors"
-	"strings"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 type GenesisState struct {
-	RewardDenom string       `json:"reward_denom"`
-	PoolAmount  sdk.DecCoins `json:"pool_amount"`
+	PoolAmount       sdk.DecCoins `json:"pool_amount"`
+	YearlyPoolAmount sdk.DecCoins `json:"yearly_pool_amount"`
+	YearNumber       int64        `json:"year_number"`
 }
 
 // DefaultGenesisState returns a default genesis state
-func DefaultGenesisState(rewardDenom string) GenesisState {
-	return GenesisState{
-		RewardDenom: rewardDenom,
-		PoolAmount:  sdk.DecCoins{},
-	}
+func DefaultGenesisState() GenesisState {
+	return GenesisState{}
 }
 
 // InitGenesis sets the initial Block Reward Pool amount for genesis.
 func InitGenesis(ctx sdk.Context, keeper Keeper, data GenesisState) {
 	if data.PoolAmount != nil {
-		keeper.SetBlockRewardsPool(ctx, data.PoolAmount)
+		keeper.SetTotalRewardPool(ctx, data.PoolAmount)
+		keeper.SetYearlyRewardPool(ctx, data.PoolAmount.QuoDec(sdk.NewDec(5)))
+	}
+
+	if data.YearlyPoolAmount != nil {
+		keeper.SetYearlyRewardPool(ctx, data.YearlyPoolAmount)
 	}
 
-	keeper.SetRewardDenom(ctx, data.RewardDenom)
+	keeper.SetYearNumber(ctx, data.YearNumber)
 }
 
 // ExportGenesis returns a GenesisState for a given context and keeper.
 func ExportGenesis(ctx sdk.Context, keeper Keeper) GenesisState {
 	return GenesisState{
-		RewardDenom: keeper.GetRewardDenom(ctx),
-		PoolAmount:  keeper.GetBlockRewardsPool(ctx),
+		PoolAmount: keeper.GetTotalRewardPool(ctx),
+		YearNumber: keeper.GetYearNumber(ctx),
 	}
 }
 
 // ValidateGenesis performs basic validation of genesis data returning an
 // error for any failed validation criteria.
 func ValidateGenesis(data GenesisState) error {
-	if len(strings.TrimSpace(data.RewardDenom)) == 0 {
-		return errors.New("transaction block reward reward denom cannot be empty")
-	}
-
 	if data.PoolAmount.Empty() {
 		return errors.New("transaction block reward pool cannot be empty")
```

Comment:
You shouldn't do some checks for the `YearNumber` data as well and possibly for the `YearlyPoolAmount` data ?

Reply:
I'm fine with having the default year number to `0` initially. Good catch for the missed `YearlyPoolReward` value checks. Added!